### DISCUSSION TITLE
ESP32-S3 and SPI2

### DISCUSSION THREAD

**Discussion by Aleks-Falcon** [micropython/micropython#10345] [ESP32]
I have board: ESP32-S3-DevKitC-1-N8R8 (Module Integrated - ESP32-S3-WROOM-1-N8R8)

I'm using the latest firmware:
GENERIC_S3_SPIRAM_OCT-20221220-unstable-v1.19.1-782-g699477d12.bin

Everything works fine, but there is a problem:

>>> hspi1 = SPI(1, 10000000)	ok

but

>>> hspi2 = SPI(2, 10000000)	ERROR!!!
	The THONNY program writes:
	PROBLEM IN THONNY'S BACK-END: Exception while handling 'execute_source' (ConnectionError: EOF).
	You may need to press "Stop/Restart" or hard-reset your MicroPython device and try again.

How do I enable and configure spi2 ?

**Comment by beyonlo** [comment]
@Aleks-Falcon please, check if has some relation with this bug #8634.

**Comment by Aleks-Falcon** [comment]
I looked at your link and assembled the firmware with a new file: ports/esp32/machine_hw_spi.c

As a result , I received:
>>> spi1  = SPI(1, baudrate=5_000_000, phase=1, sck=Pin(5), mosi=Pin(6), miso=Pin(7))	ok!	
>>> spi2  = SPI(2, baudrate=5_000_000, phase=1, sck=Pin(12), mosi=Pin(13), miso=Pin(14))	ok!

but if you write like this:
>>> spi1  = SPI(1)	ok!
>>> spi2  = SPI(2)	error!!! Process ended with exit code 1.

What is my mistake?

**Reply by beyonlo** [reply]
Hello @Aleks-Falcon 

As the ESP32 has many variations, I think that the SPI hardware pins used in the ESP32 is different than ESP32-S3/S2/C3/etc, but I'm not sure. In the ESP32 [SPI doc](https://docs.micropython.org/en/latest/esp32/quickref.html#hardware-spi-bus) shows what are default SPI pins for ESP32, but I'm not sure if that follow the same for all ESP32 microcontrollers. So set the pins and a good idea instead just `spi1 = SPI(1)` and `spi1 = SPI(1)` 

Remeber that bug #8634 was already fixed, so if you get the `Nightly builds` firmwares already has that bug fixed

**Comment by Aleks-Falcon** [comment]
Hello beyonlo
Thanks for the reply.  I have a question for you :  Have you tested and worked with this interface on ESP32-s3?

**Reply by beyonlo** [reply]
@Aleks-Falcon Yes, I'm using the `SPI` on the `ESP32-S3` without problem. I just not remember now if I'm using hardware `SPI` or Software `SPI`
